server/src/main/java/com/google/musicanalysis/api/youtube/YoutubeRequest.java

```diff
@@ -0,0 +1,77 @@
+package com.google.musicanalysis.api.youtube;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class YoutubeRequest {
+  private static final String BASE_URL = "https://www.googleapis.com/youtube/v3/";
+  private static final String apiKey = "&key=";
+
+  private String operation;
+  private ArrayList<String> parameters = new ArrayList<String>();
+
+  public YoutubeRequest(String operation, String... parameters) {
+    this.operation = operation;
+    for (String parameter : parameters) {
+      this.parameters.add(parameter);
+    }
+  }
+
+  public String getResult() throws MalformedURLException, IOException {
+
+    StringBuffer buffer = new StringBuffer();
+
+    URL url = buildUrl();
+
+    BufferedReader in = new BufferedReader(new InputStreamReader(url.openStream(), "UTF-8"));
+
+    // reading chunks of response into the buffer
+    String chunk;
+    while ((chunk = in.readLine()) != null) {
+      buffer.append(chunk);
+    }
+
+    in.close();
+
+    return buffer.toString();
+  }
+
+  private URL buildUrl() throws MalformedURLException {
+    ArrayList<String> searchOps = new ArrayList<String>(Arrays.asList("?q=", "&videoCaption="));
+    ArrayList<String> commentOps = new ArrayList<String>(Arrays.asList("?part=", "&videoId="));
```

Comment:
You should be using a map of parameters instead of an array. This is currently brittle since it relies on people remembering to pass in parameters in a specific order and for them to match up with these array lists. Instead, you can accept a map of parameters, and do something like this:
```
// javascript code. Sorry, I'm a java scrub.
function build(paramMap) {
  const parts = Object.keys(paramMap).map(key => {
    return `${encode(key)}=${encode(paramMap[key])}`;
  );
  return `${url}?${parts.join('&')}`;
}
```

Reply:
This will also future-proof your buildParams function.

Reply:
This is dope. Don't even need the buildParams function anymore.